avro_schema/frontend.lua

```diff
@@ -296,7 +296,7 @@ copy_schema = function(schema, ns, scope, open_rec, options)
                 res = scope_add_type(scope, options, name, res)
                 res.aliases = checkaliases(schema, ns, scope)
                 open_rec = open_rec or {}
-                open_rec[res] = 1
+                open_rec[name] = 1
```

Comment:
A test case that fails before this commit and succeeds after would be very helpful.